src/libpcp_web/src/query.c

```diff
@@ -4355,14 +4355,26 @@ calculate_star(int *type, pmAtomValue *l_val, pmAtomValue *r_val, pmAtomValue *r
     switch (*type) {
     case PM_TYPE_32:
 	res->l = l_val->l * r_val->l;
+	if (l_val->l != 0 && res->l / l_val->l != r_val->l) {
```

Comment:
This overflow test needs to be done before the overflow happens - I think the way you've got the unsigned integer versions is more correct.  Can you do the same checks here except using INT32_MAX?  Maybe I'm missing some subtlety here though.